public/index.html

```diff
@@ -1,46 +1,243 @@
+				"ArrowUp": () => (this.getNode('player2').direction = 0),
+				"ArrowDown": () => (this.getNode('player2').direction = 0),
+			}
+
+			console.log("***** Press Space Bar to Start *****");
 		};
 
 		app.onUpdate = function(time){
-			this.getNode('black-box').y++;
+			player1Node = this.getNode('player1');
+			player2Node = this.getNode('player2');
+			ballNode = this.getNode('ball');
 
-			if(Math.cos(this.timestamp / 100) > 0){
-				this.getNode('red-box').direction = -1;
-			}else{
-				this.getNode('red-box').direction = 1;
-			}
+			// Control Player Movement
+			handlePlayerMovement(player1Node);
+			handlePlayerMovement(player2Node);
 
-			this.getNode('red-box').x+=this.getNode('red-box').direction;
+			// Handle Ball Movement and interractions
+			// Bounce off cieling and floor
+			if (ballNode.y <= 0 || ballNode.y >= (app.height - ballNode.height)) {
+				ballNode.dy *= -1;
+			}
+			// If Ball makes contact with Player1's paddle
+			if (ballNode.x <= (player1Node.x + player1Node.width) && 
+			ballNode.x >= player1Node.x && 
+			ballNode.y + (ballNode.height / 2) >= player1Node.y && ballNode.y + (ballNode.height / 2) <= player1Node.y + player1Node.height) {
+				modifyBallVelocity(ballNode, player1Node);
+			}	
+			// If Ball makes contact with Player2's paddle
+			else if (ballNode.x + ballNode.width >= player2Node.x && ballNode.x + ballNode.width <= player2Node.x + player2Node.width &&
+			ballNode.y + (ballNode.height / 2) >= player2Node.y && ballNode.y + (ballNode.height / 2) <= player2Node.y + player2Node.height) {
+				modifyBallVelocity(ballNode, player2Node);
+			}
+			else if (ballNode.x <= 0) {
+				console.log("PLAYER 2 SCORES!!");
+				app.p2Score++;
+				console.log(`****** SCORE *****\nPlayer 1: ${app.p1Score}\nPlayer 2: ${app.p2Score}\n*****************`);
+				this.nextRound();
+			} 
+			else if (ballNode.x >= (app.width - ballNode.width)) {
+				console.log("PLAYER 1 SCORES!!");
+				app.p1Score++;
+				console.log(`****** SCORE *****\nPlayer 1: ${app.p1Score}\nPlayer 2: ${app.p2Score}\n*****************`)
+				this.nextRound();
+			} 
+			
+			ballNode.x += ballNode.dx;
+			ballNode.y += ballNode.dy;
+			
 		};
+
+		const handlePlayerMovement = (player) => {
+				// If the player is within the canvas allow to move *OR*
+			if (player.y > 0 && player.y < app.height - player.height ||
+				// If player is at top boundary, prevent from going upward *OR*
+				player.y === 0 && player.direction != -1 ||
+				// If player is at bottom boundary, prevent from going downward
+				player.y === app.height - player.height && player.direction != 1) {
+
+				player.y += player.direction * player.speed;
+			}
+			// If the player is above the top, set y to top.
+			else if (player.y < 0) {
+				player.y = 0;
+			}
+			// If the player is below the bottom, set y to bottom.
+			else if (player.y > app.height - player.height) {
+				player.y = app.height - player.height;
+			}
+		}
```

Comment:
Same with this. It should call other functions and logic should be saved inside variables so that it is much more understandable.

Example:

`if(something > 0 && somethingElse != undefined && somethingThird < 0) doSomething()`

is bad
```
if(canDoSomething()) doSomething()

function canDoSomething() {
  if(something < 0) return false
  if(somethingElse == undefined) return false
  if(somethingThird > 0) return false
  return true
}
```
is good 

or 

```
const canDoFirstThing = something > 0
const secondThingExists = somethingElse != undefined
const canDoThirdThing = somethingThird < 0
`if(canDoFirstThing && secondThingExists && canDoThirdThing) doSomething()`
```
is good

Reply:
Giving such a name `const canDoFirstThing = something > 0` makes the code more explicit thus no need for description.

Reply:
I agree with this method that you have laid out. I went through and updated my conditionals and functions to have meaningful names so as to make it clear what the code is doing.